code/modules/vehicles/segway.dm

```diff
@@ -0,0 +1,221 @@
+	if(istype(W, /obj/item/weapon/key/segwaykey))
+		if(!key)
+			user.drop_from_inventory(W,src)
+			key = W
+			verbs += /obj/vehicle/train/cargo/engine/verb/remove_key
+		return
+	..()
+
+/obj/vehicle/segway/turn_on()
+	if(!key)
+		return
+	else
+		..()
+		update_stats()
+
+		verbs -= /obj/vehicle/segway/verb/stop_engine
+		verbs += /obj/vehicle/segway/verb/start_engine
+
+		if(on)
+			verbs += /obj/vehicle/segway/verb/stop_engine
+		else
+			verbs += /obj/vehicle/segway/verb/start_engine
+
+/obj/vehicle/train/cargo/engine/turn_off()
+	..()
+
+	verbs += /obj/vehicle/segway/verb/stop_engine
+	verbs += /obj/vehicle/segway/verb/start_engine
+
+	if(!on)
+		verbs += /obj/vehicle/segway/verb/start_engine
+	else
+		verbs += /obj/vehicle/segway/verb/stop_engine
+		verbs += /obj/vehicle/segway/verb/start_engine
+
+
+
+/obj/vehicle/segway/verb/start_engine()
+	set name = "Start engine"
+	set category = "Vehicle"
+	set src in view(0)
+
+	if(!istype(usr, /mob/living/carbon/human))
+		return
+
+	if(on)
+		usr << "The engine is already running."
+		return
+
+	turn_on()
+	if (on)
+		usr << "You start [src]'s engine."
+		playsound(src, 'sound/misc/bike_start.ogg', 100, 1)
+	else
+		if(cell.charge < charge_use)
+			usr << "[src] is out of power."
+		else
+			usr << "[src]'s engine won't start."
+
+/obj/vehicle/segway/verb/stop_engine()
+	set name = "Stop engine"
+	set category = "Vehicle"
+	set src in view(0)
+
+	if(!istype(usr, /mob/living/carbon/human))
+		return
+
+	if(!on)
+		usr << "The engine is already stopped."
+		return
+
+	turn_off()
+	if (!on)
+		usr << "You stop [src]'s engine."
+
+
+/obj/vehicle/segway/verb/siren()
+	set name = "Hail Suspect"
+	set category = "Vehicle"
+	set src in view(0)
+
+	if(usr.incapacitated()) return
+
+	if(!on)
+		usr.visible_message("\The [usr] attempts to use [src]'s siren but it won't work with the engine off.", "You attempt to use \the [src]'s siren with the engine off, it won't work however.", "You hear a thunk.")
+		return
+	else
+		playsound(get_turf(src), 'sound/voice/halt.ogg', 100, 1, vary = 0)
+
+
+/obj/vehicle/segway/verb/kickstand()
+	set name = "Toggle Kickstand"
+	set category = "Vehicle"
+	set src in view(0)
+
+	if(usr.incapacitated()) return
+
+	if(kickstand)
+		usr.visible_message("\The [usr] puts up \the [src]'s kickstand.", "You put up \the [src]'s kickstand.", "You hear a thunk.")
+		playsound(src, 'sound/misc/bike_stand_up.ogg', 50, 1)
+	else
+		if(isturf(loc))
+			var/turf/T = loc
+			if (T.is_hole)
+				usr << "<span class='warning'>You don't think kickstands work here.</span>"
+				return
+		usr.visible_message("\The [usr] puts down \the [src]'s kickstand.", "You put down \the [src]'s kickstand.", "You hear a thunk.")
+		playsound(src, 'sound/misc/bike_stand_down.ogg', 50, 1)
+		if(pulledby)
+			pulledby.stop_pulling()
+
+	kickstand = !kickstand
+	anchored = (kickstand || on)
+
+/obj/vehicle/segway/load(var/atom/movable/C)
+	var/mob/living/M = C
+	if(!istype(C)) return 0
```

Comment:
You'll want `if (!istype(M))`. Further, don't one-line code like this, it hurts readability.